src/libostree/ostree-sysroot-deploy.c

```diff
@@ -1700,21 +1700,29 @@ install_deployment_kernel (OstreeSysroot   *sysroot,
   g_autofree char * boot_relpath = g_strconcat ("/", bootcsumdir, "/", kernel_layout->kernel_namever, NULL);
   ostree_bootconfig_parser_set (bootconfig, "linux", boot_relpath);
 
+  val = ostree_bootconfig_parser_get (bootconfig, "options");
+  g_autoptr(OstreeKernelArgs) kargs = _ostree_kernel_args_from_string (val);
+
   if (kernel_layout->initramfs_namever)
     {
       g_autofree char * boot_relpath = g_strconcat ("/", bootcsumdir, "/", kernel_layout->initramfs_namever, NULL);
       ostree_bootconfig_parser_set (bootconfig, "initrd", boot_relpath);
     }
-
-  val = ostree_bootconfig_parser_get (bootconfig, "options");
+  else
+    {
+      g_autofree char *prepare_root_arg = NULL;
+      prepare_root_arg = g_strdup_printf ("init=/ostree/boot.%d/%s/%s/%d/usr/lib/ostree/ostree-prepare-root",
```

Comment:
This path should use `LIBEXECDIR` I think.

Reply:
I don't think so.  See fd1a044d6e75b336ea880530979067926c9b0a09.  I think the idea is that `ostree-prepare-root` needs to be at a fixed location in the rootfs otherwise a different version of ostree with different configure flags couldn't successfully deploy a rootfs with standard configure flags.

Reply:
Yep, I even agreed with this in https://github.com/ostreedev/ostree/pull/449#issuecomment-238588405 - though for a different reason; I'm not sure how realistic it is to worry about different configure flags.